jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3035,6 +3043,77 @@ public MethodHandle run() {
 	return valueMethod;
 }
 
+private MetadataCache getMetadataCache() {
+	if (metadataCache == null) {
+		try {
+			Field field = Class.class.getDeclaredFieldInternal("metadataCache", false); //$NON-NLS-1$
+			metadataCacheOffset = getUnsafe().objectFieldOffset(field);
```

Comment:
Maybe jdk11+ should use objectFieldOffset(Class, String).

Reply:
Updated for `JDK11+`, the day `JDK8` becomes obsoleted will be remarkable though.